docs: Add CSI volume snapshot documentation (#17085)
canonical/lxd: doc/reference/driver_csi.md

`node.csiLivenessProbe.image.tag`                  | string | Chart release dependent                                 | CSI liveness probe image tag
`node.csiLivenessProbe.image.pullPolicy`           | string | `IfNotPresent`                                          | CSI liveness probe image pull policy
`node.csiLivenessProbe.resources`                  | object | `{}`                                                    | CSI liveness probe resource limits and requests

> `snapshotter.enabled`                              | bool   | `false`                                                 | Whether to enable support for volume snapshots. If enabled, CSI snapshot controller is deployed along with CSI driver itself.

Reviewer: Would be nice to include an example of how to install the helm chart with that option on

> `snapshotter.enabled`                              | bool   | `false`                                                 | Whether to enable support for volume snapshots. If enabled, CSI snapshot controller is deployed along with CSI driver itself.

Reviewer 2: ```suggestion
`snapshotter.enabled`                              | bool   | `false`                                                 | Whether to enable support for volume snapshots. If enabled, the CSI snapshot controller is deployed along with the CSI driver.
```

> `snapshotter.installCRDs`                          | bool   | `true`                                                  | Whether to install required volume snapshot CRDs. The value is ignored if `snapshotter.enabled` is false. Should be disabled if CRDs are installed manually or by other CSI driver.

Reviewer 2: Let's word it less ambiguously, so that it's clear that the user is expected to perform some action:

```suggestion
`snapshotter.installCRDs`                          | bool   | `true`                                                  | Whether to install required volume snapshot CRDs. If CRDs are installed manually or by other CSI drivers, set this to `false` to avoid conflicts with other drivers . The value is ignored if `snapshotter.enabled` is false. 
```
